src/main/java/mate/jdbc/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package mate.jdbc.service.impl;
+
+import java.util.List;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.service.CarService;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        if (car == null) {
+            throw new DataProcessingException("Can't create car from null");
+        }
```

Comment:
lets replace **all** null checks with  [Objects.requireNonNull()](https://stackoverflow.com/questions/45632920/why-should-one-use-objects-requirenonnull), wich is more standardised and compact.

you can also add a JetBrains `NotNull` annotation to the needed parameter in the method signature to make it even more explicit and shown in the hint when you hover your mouse over the method name if you want

Reply:
add anotation sounds the best, thanks!